src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/RetrieveOrganisationByOrgIdTest.java

```diff
@@ -0,0 +1,131 @@
+
+import io.restassured.specification.RequestSpecification;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import lombok.extern.slf4j.Slf4j;
+import net.serenitybdd.junit.spring.integration.SpringIntegrationSerenityRunner;
+
+import org.assertj.core.api.Assertions;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.professionalapi.controller.request.NewUserCreationRequest;
+import uk.gov.hmcts.reform.professionalapi.controller.request.OrganisationCreationRequest;
+import uk.gov.hmcts.reform.professionalapi.controller.request.UserCreationRequest;
+
+
+@RunWith(SpringIntegrationSerenityRunner.class)
+@ActiveProfiles("functional")
+@Slf4j
+public class RetrieveOrganisationByOrgIdTest extends AuthorizationFunctionalTest {
+
+    private RequestSpecification bearerTokenForUser;
+    private String orgIdentifier;
+    private String email = randomAlphabetic(10) + "@pbasearch.test".toLowerCase();
+    private String lastName = "someLastName";
+    private String firstName = "someName";
+
+    public RequestSpecification generateBearerTokenForUser(String roleUnderTest, String... otherRoles) {
+        Map<String, Object> response = professionalApiClient.createOrganisation();
+        orgIdentifier = (String) response.get("organisationIdentifier");
+        professionalApiClient.updateOrganisation(orgIdentifier, hmctsAdmin);
+
+        List<String> userRoles = Arrays.stream(otherRoles).collect(Collectors.toList());
+        userRoles.add(roleUnderTest);
+
+        bearerTokenForUser = professionalApiClient.getMultipleAuthHeadersExternal(roleUnderTest, firstName, lastName, email);
+
+        NewUserCreationRequest userCreationRequest = aNewUserCreationRequest()
+                .firstName(firstName)
+                .lastName(lastName)
+                .email(email)
+                .roles(userRoles)
+                .jurisdictions(createJurisdictions())
+                .build();
+
+        professionalApiClient.addNewUserToAnOrganisation(orgIdentifier, hmctsAdmin, userCreationRequest, HttpStatus.CREATED);
+
+        return bearerTokenForUser;
+    }
+
+    @Test
+    public void rdcc117_ac1_pui_finance_manager_can_retrieve_organisation_by_orgIdentifier_for_external() {
+        Map<String, Object> response = professionalApiClient.retrievePbaAccountsForAnOrganisationExternal(HttpStatus.OK, generateBearerTokenForUser(puiFinanceManager));
+        assertThat(response.get("paymentAccount")).asList().hasSize(3);
+        responseValidate(response);
+    }
+
+    @Test
+    public void rdcc117_ac2_pui_organisation_manager_can_retrieve_organisation_by_orgIdentifier_for_external() {
+        Map<String, Object> response = professionalApiClient.retrievePbaAccountsForAnOrganisationExternal(HttpStatus.OK, generateBearerTokenForUser(puiOrgManager));
```

Comment:
Can you please delete the three retrievePbaAccountsForAnOrganisationExternal functional test which already present in the RetrievePaymentAccountTest

Reply:
Yes can be removed in next release.

Reply:
Created tech debt for this

Reply:
https://tools.hmcts.net/jira/browse/RDCC-1680